File: 49-Group-Anagrams/49.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        Dictionary<string, IList<string> > res = new Dictionary<string, IList<string> >();
        foreach(string s in strs){
            int[] count = new int[26];
            foreach(char c in s){
                count[c - 'a']++;
            }
            string key = string.Join(",", count);
            if(!res.ContainsKey(key)){
                res[key] = new List<string>();
            }
            res[key].Add(s);
        }
        return res.Values.ToList();
    }
}

File: Arrays and Hashing/49-Group-Anagrams/49.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const res = {};
    for(let i = 0; i < strs.length; i++){
        const count = new Array(26).fill(0);
        for(let j = 0; j < strs[i].length; j++){
            count[strs[i][j].charCodeAt(0) - 'a'.charCodeAt(0)] += 1;
        }
        const key = count.join(',');
        if(!res[key]){
            res[key] = [];
        }
        res[key].push(strs[i]);
    }
    return Object.values(res);
};